### PR TITLE
Phoneme interpretor

### PR DESCRIPTION
**Pull request by ngoguened**

### CODE REVIEW

File: phoneme_interpreter_tests.py [hunk 1]
```diff
@@ -9,9 +11,13 @@
 class InterpretersTest(unittest.TestCase):
     """Test each implemented interpreter"""
     def test_interpret(self):
-        interpreters = [DictPhonemeInterpreter()]
+        interpreters:list[PhonemeInterpreterInterface] = [
+            TestPhonemeInterpreter(), DictPhonemeInterpreter()
+            ]
         for interpreter in interpreters:
-            assert interpreter.interpret([Phoneme(phoneme=PhonemeEnums.h), Phoneme(phoneme=PhonemeEnums.aɪ)])
+            assert interpreter.interpret([
```
Comment (healfgoguen) [inline phoneme_interpreter_tests.py:18]: This should be more specific and validate that the returned words are as expected.

File: test_phoneme_interpreter.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from phoneme_interpreter_interface import PhonemeInterpreterInterface
+
+class TestPhonemeInterpreter(PhonemeInterpreterInterface):
+
+    def interpret(self, phonemes):
+        return ["fee", "fie", "foe", "fum"] if phonemes else []
```
Comment (healfgoguen) [inline test_phoneme_interpreter.py:6]: This should return something like ["hi", "high"] if phonemes = ["h", "ai"], and otherwise return empty.

File: phoneme_interpreter_tests.py [hunk 3]
```diff
@@ -9,9 +11,13 @@
...
+                Phoneme(phoneme=PhonemeEnums.h), Phoneme(phoneme=PhonemeEnums.aɪ)
+                ])
```
Comment (healfgoguen) [inline phoneme_interpreter_tests.py:20]: Also, there should be a test that the returned list is empty if there's a phoneme list that doesn't correspond to a word.

File: phoneme_interpreter_tests.py [hunk 4]
```diff
@@ -9,9 +11,13 @@
 class InterpretersTest(unittest.TestCase):
     """Test each implemented interpreter"""
     def test_interpret(self):
-        interpreters = [DictPhonemeInterpreter()]
+        interpreters:list[PhonemeInterpreterInterface] = [
+            TestPhonemeInterpreter(), DictPhonemeInterpreter()
+            ]
         for interpreter in interpreters:
```
Comment (healfgoguen) [inline phoneme_interpreter_tests.py:17]: It would be better to only test the TestPhonemeInterpreter, which has predictable behavior.
